File: src/meshres.f90
subroutine meshres(VB,nvb,IT,nit,res)

  integer :: nvb,nit,i,j,IT(3,nit)
  real*8 :: VB(3,nvb),res,tmp(3)

  do i = 1,nit
     tmp = VB(1:3,IT(1,i))-VB(1:3,IT(2,i))
     res= res+sqrt(dot_product(tmp,tmp))
     tmp = VB(1:3,IT(1,i))-VB(1:3,IT(3,i))
     res= res+sqrt(dot_product(tmp,tmp))
     tmp = VB(1:3,IT(2,i))-VB(1:3,IT(3,i))
     res= res+sqrt(dot_product(tmp,tmp))
  end do
  res = res/(nit*3)

end subroutine meshres

File: src/meshres.cpp
#include <RcppArmadillo.h>
using namespace Rcpp;
using namespace std;
using namespace arma;

RcppExport SEXP meshresCpp(SEXP vb_, SEXP it_) {
  try {
        
    arma::mat vbA = Rcpp::as<arma::mat>(vb_);
    //mat vbA(vb.begin(),vb.nrow(),vb.ncol());
    imat itA = as<arma::imat>(it_);
    int nit = itA.n_cols;
    //imat itA(it.begin(),it.nrow(),it.ncol());
    vec tmp(3);
    double res = 0.0;
    for (int i=0; i < nit;++i) {
      tmp = vbA.col(itA(0,i))-vbA.col(itA(1,i));
      res += sqrt(dot(tmp,tmp));
      tmp = vbA.col(itA(0,i))-vbA.col(itA(2,i));
      res += sqrt(dot(tmp,tmp));
      tmp = vbA.col(itA(1,i))-vbA.col(itA(2,i));
      res += sqrt(dot(tmp,tmp));
    }
    res /= nit*3;
    return wrap(res);
  } catch (std::exception& e) {
    ::Rf_error( e.what());
  } catch (...) {
    ::Rf_error("unknown exception");
  }
}
